### PR TITLE
SS-1089 Add connection pooling

### PR DESCRIPTION
**Pull request by hamzaimran08**
## Description

This PR refers to [SS-1089](https://scilifelab.atlassian.net/browse/SS-1089). Added connection pooling support by upgrading Django to version 5.1.1 and then adding pooling settings in `settings.py`. Also updated other packages especially psycopg to version 3 to use connection pooling.

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [ ] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [ ] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [x] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -18,27 +18,27 @@ packages = []
 [tool.poetry.dependencies]
 python = "^3.10.0"
 
-django = "==5.0.2"
-django-celery-beat = "==2.6.0"
-django-compressor = "==4.4"
-django-cors-headers = "==4.3.1"
+django = "==5.1.1"
```
Comment (churnikov) [inline pyproject.toml:21]: You've updated this file, but did not push `poetry.lock` file. You can run `poetry lock --no-update` to update the file

File: Dockerfile [hunk 2]
```diff
@@ -6,7 +6,6 @@ WORKDIR /app
 ARG DISABLE_EXTRAS=false
 
 COPY pyproject.toml ./
-COPY poetry.lock ./
```
Comment (churnikov) [inline Dockerfile:9]: Why did you remove it? It's needed for the environment reproducibility

**Review by churnikov — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (hamzaimran08) [reply pyproject.toml:21]: This is what I was missing. Thanks.

File: Dockerfile (hunk 2 above)
Comment (hamzaimran08) [reply Dockerfile:9]: I removed it to test if CI is able to build the image. It was failing on the writing lock file part. It seems to work now that I have updated the lock file with `poetry lock --no-update`.
